src/components/services/EdgeCoreManager.tsx

```diff
@@ -67,8 +67,55 @@ const crashReporter: EdgeCrashReporter = {
     })
   },
   logCrash(event) {
-    const eventString = JSON.stringify(event, null, 2)
-    captureException(eventString, { level: 'fatal' })
+    // Index the crash error by the source and original error name:
+    const error = new Error(`${event.source}: ${String(event.error)}`)
+    // All of these crash errors are grouped together using this error name:
+    error.name = 'EdgeCrashLog'
+
+    captureException(error, scope => {
+      scope.setLevel('fatal')
+
+      const context: Record<string, unknown> = {}
+      addMetadataToContext(context, event.metadata)
+      scope.setContext('Edge Crash Metadata', context)
+
+      return scope
+    })
+  }
+}
+
+/**
+ * Recursively adds metadata to a context object.
+ *
+ * @param context The context object to which to add metadata
+ * @param metadata The metadata object to add to the context
+ * @param prefixKeys optional prefix keys to add to each entry
+ * @returns void (modifies context object)
+ */
+function addMetadataToContext(context: Record<string, unknown>, metadata: object, prefixKeys: string[] = []): void {
+  for (const [key, value] of Object.entries(metadata)) {
+    const allKeys = [...prefixKeys, key]
+    const fullKey = allKeys.join(' > ')
+
+    // Serialize error objects manually because JSON.stringify doesn't
+    // include all the error properties:
+    if (value instanceof Error) {
+      context[fullKey] = `${value.name}: ${value.message}`
+      if (value.stack != null) {
+        // Include the stack trace with indentation:
+        context[fullKey] += `\n    ${value.stack.replace(/\n/g, '\n    ')}`
+      }
+      continue
+    }
+
+    // Recurse over objects with added prefix keys:
+    if (typeof value === 'object' && value !== null) {
+      addMetadataToContext(context, value, allKeys)
+      continue
+    }
```

Comment:
Should we just `JSON.stringify` instead of recursing? Imagine you have something like this:

```
{
  fromCoin: { tokenId: 'asf', pluginId: 'asdf' },
  toCoin: { tokenId: 'dda', pluginId: 'asafadf' }
}
```

The fields in `toCoin` would stomp over the fields in `fromCoin`, and you would lose information. If we use `JSON.stringify` on the values, then you would get:

```
{
  fromCoin: '{"tokenId":"asf","pluginId":"asdf"}',
  toCoin: '{"tokenId":"dda","pluginId":"asafadf"}'
}
```

This would be more usable, I think.

Reply:
The reason why: JSON strings are big and when metadata fields are too big, they're filtered out by Sentry. So, I flattened the JSON objects into separate fields. Each field contains the parent key in it's name so there shoulnd't be possibility of key stomping